core/arch/arm/kernel/generic_boot.c

```diff
@@ -552,6 +555,37 @@ static int add_optee_res_mem_dt_node(void *fdt)
 	return 0;
 }
 
+static void read_optee_pseudo_device_node(void *fdt)
+{
+	char s[33];
+	int offs, n = 0, m, len;
+	const uint32_t *p;
+
+	offs = fdt_path_offset(fdt, "/firmware/optee");
+	if (offs < 0)
+		return;
+
+	p = fdt_getprop(fdt, offs, "pseudo-device-id", &len);
+	if (!p || !len)
+		return;
+
+	if (len > (int)sizeof(pseudo_device_id))
+		len = sizeof(pseudo_device_id);
+
+	memcpy(pseudo_device_id, p, len);
+	pseudo_device_id_len = len;
+
+	if (len > 16)
```

Comment:
I tried to follow this but it's a mess, DHEXDUMP doesn't quite do what you seem to think it does here.  The end result is

```
INFO:    TEE-CORE: secure-device-id: len 4: 
INFO:    TEE-CORE: 000000003f039f28  08 35 d2 ac 
```
ie, it starts a newline, looks like it dumps line number, function name, and insists to dump the VA of the start of the buffer.

What do you want to do... go back to old way protected by preprocessor check for INFO level OK?  Or add like dump_raw()?

Both the truncation and the info is because I want to be able to confirm that users have unique serials without having a debug spew, and I don't need to dump 160 bytes each time to confirm that.